src/network/core/tcp_coordinator.h

```diff
@@ -23,15 +23,22 @@
  * GC     -> packets from Game Coordinator to either Client or Server.
  * SERVER -> packets from Server to Game Coordinator.
  * CLIENT -> packets from Client to Game Coordinator.
+ * SERCLI -> packets from either the Server or Client to Game Coordinator.
  **/
 enum PacketCoordinatorType {
-	PACKET_COORDINATOR_GC_ERROR,        ///< Game Coordinator indicates there was an error.
-	PACKET_COORDINATOR_SERVER_REGISTER, ///< Server registration.
-	PACKET_COORDINATOR_GC_REGISTER_ACK, ///< Game Coordinator accepts the registration.
-	PACKET_COORDINATOR_SERVER_UPDATE,   ///< Server sends an set intervals an update of the server.
-	PACKET_COORDINATOR_CLIENT_LISTING,  ///< Client is requesting a listing of all public servers.
-	PACKET_COORDINATOR_GC_LISTING,      ///< Game Coordinator returns a listing of all public servers.
-	PACKET_COORDINATOR_END,             ///< Must ALWAYS be on the end of this list!! (period).
+	PACKET_COORDINATOR_GC_ERROR,              ///< Game Coordinator indicates there was an error.
+	PACKET_COORDINATOR_SERVER_REGISTER,       ///< Server registration.
+	PACKET_COORDINATOR_GC_REGISTER_ACK,       ///< Game Coordinator accepts the registration.
+	PACKET_COORDINATOR_SERVER_UPDATE,         ///< Server sends an set intervals an update of the server.
+	PACKET_COORDINATOR_CLIENT_LISTING,        ///< Client is requesting a listing of all public servers.
+	PACKET_COORDINATOR_GC_LISTING,            ///< Game Coordinator returns a listing of all public servers.
+	PACKET_COORDINATOR_CLIENT_CONNECT,        ///< Client wants to connect to a server based on an invite code.
+	PACKET_COORDINATOR_GC_CONNECTING,         ///< Game Coordinator informs the client of the token assigned to the connection attempt.
+	PACKET_COORDINATOR_SERCLI_CONNECT_FAILED, ///< Client/server tells the Game Coordinator the current connection attempt failed.
```

Comment:
Won't it be cleaner/clearer to just have a separate packet for a server failure? Maybe you want to sent other information in such cases? The current code asserts on not being a server before sending this packet.

Reply:
This name is picked not so much for this PR, but for the future STUN PR. I didn't want to rename the packet, so I already sneaked in how STUN is going to work :)

Basically, this packet is send back as a response to the GC asking you to connect to somewhere. For direct-connect, what this PR only support, it is only the client that can do this. But with STUN, the GC sends both the server and client a "please connect here" request. They both answer with a "failure" when they cannot. They are designed to be identical in payload, as for STUN it doesn't matter who is server and who is client. As such, they will always be identical. And so I put them in a single packet.

But I agree in this PR it looks a bit weird, as it is never used from the server point of view. But renaming a packet is also a bit meh in the long run. A bit of a balance here, I guess. But I rather keep this name so it makes sense after the STUN PR. Unless you really want me to, I can rename it `CLIENT` here to rename it back into `SERCLI` with the STUN PR :)

Other suggestions are of course welcome, but having this in two packets will be more confusing imo :)